pandas/tests/test_tseries.py

```diff
@@ -737,6 +737,17 @@ def test_get_period_field_raises_on_out_of_range(self):
     def test_get_period_field_array_raises_on_out_of_range(self):
         self.assertRaises(ValueError, period.get_period_field_arr, -1, np.empty(1), 0)
 
+class TestDaysInMonth(tm.TestCase):
+    def test_day_not_in_month_coerce_true(self):
+        self.assertTrue(isnull(to_datetime('2015-02-29', coerce=True)))
+        self.assertTrue(isnull(to_datetime('2015-02-29', format="%Y-%m-%d", coerce=True)))
+        self.assertTrue(isnull(to_datetime('2015-02-32', format="%Y-%m-%d", coerce=True)))
+        self.assertTrue(isnull(to_datetime('2015-04-31', format="%Y-%m-%d", coerce=True)))
+    def test_day_not_in_month_coerce_false(self):
+        self.assertRaises(ValueError, to_datetime, '2015-02-29', format="%Y-%m-%d", coerce=False)
```

Comment:
This looks correct
```
In [10]: pd.to_datetime('2015-02-29', errors='ignore',  coerce=False)
Out[10]: '2015-02-29'
```
I would expect the same from this but get a ValueError.
```
In [12]: pd.to_datetime('2015-02-29', errors='ignore', format="%Y-%m-%d", coerce=False)
....
ValueError: day is out of range for month
```
adding format="%Y-%m-%d" should not change the output.
Is this correct and also needs to be fixed?